<a href="https://colab.research.google.com/github/Aman151295/Hypothesis_Testing_Problems/blob/main/Hypothesis_testing.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [3]:
#Q1

from google.colab import files
uploaded = files.upload()


Saving Cutlets.csv to Cutlets.csv


In [4]:
!pip install scipy

import pandas as pd
from scipy import stats
import io

Cutlet_Data = pd.read_csv(io.BytesIO(uploaded['Cutlets.csv']))
print(Cutlet_Data)

    Unit A  Unit B
0   6.8090  6.7703
1   6.4376  7.5093
2   6.9157  6.7300
3   7.3012  6.7878
4   7.4488  7.1522
5   7.3871  6.8110
6   6.8755  7.2212
7   7.0621  6.6606
8   6.6840  7.2402
9   6.8236  7.0503
10  7.3930  6.8810
11  7.5169  7.4059
12  6.9246  6.7652
13  6.9256  6.0380
14  6.5797  7.1581
15  6.8394  7.0240
16  6.5970  6.6672
17  7.2705  7.4314
18  7.2828  7.3070
19  7.3495  6.7478
20  6.9438  6.8889
21  7.1560  7.4220
22  6.5341  6.5217
23  7.2854  7.1688
24  6.9952  6.7594
25  6.8568  6.9399
26  7.2163  7.0133
27  6.6801  6.9182
28  6.9431  6.3346
29  7.0852  7.5459
30  6.7794  7.0992
31  7.2783  7.1180
32  7.1561  6.6965
33  7.3943  6.5780
34  6.9405  7.3875


In [5]:

Unit_A = Cutlet_Data['Unit A']
print(Unit_A)

Unit_B = Cutlet_Data['Unit B']
print(Unit_B)

0     6.8090
1     6.4376
2     6.9157
3     7.3012
4     7.4488
5     7.3871
6     6.8755
7     7.0621
8     6.6840
9     6.8236
10    7.3930
11    7.5169
12    6.9246
13    6.9256
14    6.5797
15    6.8394
16    6.5970
17    7.2705
18    7.2828
19    7.3495
20    6.9438
21    7.1560
22    6.5341
23    7.2854
24    6.9952
25    6.8568
26    7.2163
27    6.6801
28    6.9431
29    7.0852
30    6.7794
31    7.2783
32    7.1561
33    7.3943
34    6.9405
Name: Unit A, dtype: float64
0     6.7703
1     7.5093
2     6.7300
3     6.7878
4     7.1522
5     6.8110
6     7.2212
7     6.6606
8     7.2402
9     7.0503
10    6.8810
11    7.4059
12    6.7652
13    6.0380
14    7.1581
15    7.0240
16    6.6672
17    7.4314
18    7.3070
19    6.7478
20    6.8889
21    7.4220
22    6.5217
23    7.1688
24    6.7594
25    6.9399
26    7.0133
27    6.9182
28    6.3346
29    7.5459
30    7.0992
31    7.1180
32    6.6965
33    6.5780
34    7.3875
Name: Unit B, dtype: float64


In [27]:
# 2-sample 2-tail ttest:   stats.ttest_ind(array1,array2)     # ind -> independent samples
# Assumptions

#Ho = There is no difference between diameters of Cutlets from two Units      Mean of diameters from Unit A  = Mean of Diameters from Unit B 
#H1  = There is some difference between diameters of Cutlets from two Units   Mean of diameters from Unit A != Mean of Diameters from Unit B 


p_value=stats.ttest_ind(Unit_A,Unit_B)
p_value

#Here in above Evaluation we have T statistical value as index 0 &  corresponding P value as index 1, so to get only P value we use below command 
p_value[1]


0.4722394724599501

As we have been given a significance level Alpha = 0.05
We can compare P value = 0.47 with Alpha = 0.05, As P is gretaer than Alpha(Significance Value = 0.05)....

We Can Conclude In a way that we are accepting Null Hypothesis And we can say that

 **There is no difference between diameters of Cutlets from two Units **

**Q2.** **We have been given TAT - turn around reports for 4 hospitals, We are supposed to analyze the data as we have 4 different group samples....We can perform One way annova test**

In [6]:
from google.colab import files
uploaded = files.upload()

Saving LabTAT.csv to LabTAT.csv


In [11]:
# We will use the f_oneway() function from the SciPy library
import io
Lab_Reports = pd.read_csv(io.BytesIO(uploaded['LabTAT.csv']))
print(Lab_Reports)
    
#find all the groups 

Different_Labs = Lab_Reports.columns
print(Different_Labs)
    
F_test = stats.f_oneway(Lab_Reports.iloc[:,0],Lab_Reports.iloc[:,1],Lab_Reports.iloc[:,2],Lab_Reports.iloc[:,3])
print(F_test)

     Laboratory 1  Laboratory 2  Laboratory 3  Laboratory 4
0          185.35        165.53        176.70        166.13
1          170.49        185.91        198.45        160.79
2          192.77        194.92        201.23        185.18
3          177.33        183.00        199.61        176.42
4          193.41        169.57        204.63        152.60
..            ...           ...           ...           ...
115        178.49        170.66        193.80        172.68
116        176.08        183.98        215.25        177.64
117        202.48        174.54        203.99        170.27
118        182.40        197.18        194.52        150.87
119        182.09        215.17        221.49        162.21

[120 rows x 4 columns]
Index(['Laboratory 1', 'Laboratory 2', 'Laboratory 3', 'Laboratory 4'], dtype='object')
F_onewayResult(statistic=118.70421654401437, pvalue=2.1156708949992414e-57)


From Above results, we have F Statistical score = 118.704 & Associated P value = 2.11...... Hence P value is greater than Significance value 0.05.

**We fail to reject the null hypothesis.**

We can conclude by retaining the Null Hypothesis which states that there is no differece average TAT  of laboratories.

**Q3.** Sales of product in four different regions is tabulated for males and frmales. Find if female buyers and male buyers are similar across regions or not

In [2]:
from google.colab import files
uploaded = files.upload()


Saving BuyerRatio.csv to BuyerRatio.csv


In [3]:
!pip install scipy

import pandas as pd
from scipy import stats
import io
import numpy as np
from scipy.stats import chi2_contingency


Sale_Report = pd.read_csv(io.BytesIO(uploaded['BuyerRatio.csv']))
print(Sale_Report)
  

# For defining the corresponding array table for dataset

Table = np.array(Sale_Report)
print(Table)

Array_Table = np.delete(Table, 0, 1)
print(Array_Table)

# Chi test for observed table, will return a test statistics, the p-value, degrees of freedom and expected table
chi2_contingency(Array_Table)

  Observed Values  East  West  North  South
0           Males    50   142    131     70
1         Females   435  1523   1356    750
[['Males' 50 142 131 70]
 ['Females' 435 1523 1356 750]]
[[50 142 131 70]
 [435 1523 1356 750]]


(1.5959455386610577,
 0.6603094907091882,
 3,
 array([[  42.76531299,  146.81287862,  131.11756787,   72.30424052],
        [ 442.23468701, 1518.18712138, 1355.88243213,  747.69575948]]))

**Q4.** Tele Call uses 4 centers around the globe. In audit if there are error form that are being rendered as defective and has to be reworked before processing.

The MAnager wants to  check whether the defective % varies by center. 

#Ho = Defective % are not getting varied by centres 

#H1  = Defective % are getting varied by centres 

In [30]:
from google.colab import files
uploaded = files.upload()

Saving Costomer+OrderForm.csv to Costomer+OrderForm.csv


In [62]:
import pandas as pd
from scipy import stats
import io
import numpy as np
from scipy.stats import chi2_contingency

Audit_Report = pd.read_csv(io.BytesIO(uploaded['Costomer+OrderForm.csv']))


India_Report = Audit_Report['India'].value_counts()
print('Audit Report of India : ' , India_Report)

Phillippines_Report = Audit_Report['Phillippines'].value_counts()
print('Audit Report of Phillippines : ' , Phillippines_Report)

Indonesia_Report = Audit_Report['Indonesia'].value_counts()
print('Audit Report of Indonesia : ' , Indonesia_Report)

Malta_Report = Audit_Report['Malta'].value_counts()
print('Audit Report of Malta : ' , Malta_Report)

# preparing Observed table 

observed = np.array([[280,271,267,269],[20,29,33,31]])
print(observed)

# Chi test for observed table will return us P value

from scipy.stats import chi2_contingency

Test_results = chi2_contingency(observed)
print('Statistical result : ' '\n', Test_results)



print('P value :', chi2_contingency(observed)[1])

Audit Report of India :  Error Free    280
Defective      20
Name: India, dtype: int64
Audit Report of Phillippines :  Error Free    271
Defective      29
Name: Phillippines, dtype: int64
Audit Report of Indonesia :  Error Free    267
Defective      33
Name: Indonesia, dtype: int64
Audit Report of Malta :  Error Free    269
Defective      31
Name: Malta, dtype: int64
[[280 271 267 269]
 [ 20  29  33  31]]
Statistical result : 
 (3.858960685820355, 0.2771020991233135, 3, array([[271.75, 271.75, 271.75, 271.75],
       [ 28.25,  28.25,  28.25,  28.25]]))
P value : 0.2771020991233135


On comparing above P Value from Alpha = 0.05, we found P value > than Alpha..
Hence, We failed to reject null Hypothesis.


**Ho = Defective % are not getting varied by centres**


In [28]:
import pandas as pd
import seaborn as sns



Df = sns.load_dataset('iris')
print('Top 5 : ' + str(Df.head()))

DF2 = Df[['petal_length' , 'species' ]]
print('New :', DF2)

group = pd.unique(DF2.species.values)
print('group;', group)

Result = {grp : DF2['petal_length'] for grp in group}
print(Result)

Top 5 :    sepal_length  sepal_width  petal_length  petal_width species
0           5.1          3.5           1.4          0.2  setosa
1           4.9          3.0           1.4          0.2  setosa
2           4.7          3.2           1.3          0.2  setosa
3           4.6          3.1           1.5          0.2  setosa
4           5.0          3.6           1.4          0.2  setosa
New :      petal_length    species
0             1.4     setosa
1             1.4     setosa
2             1.3     setosa
3             1.5     setosa
4             1.4     setosa
..            ...        ...
145           5.2  virginica
146           5.0  virginica
147           5.2  virginica
148           5.4  virginica
149           5.1  virginica

[150 rows x 2 columns]
group; ['setosa' 'versicolor' 'virginica']
{'setosa': 0      1.4
1      1.4
2      1.3
3      1.5
4      1.4
      ... 
145    5.2
146    5.0
147    5.2
148    5.4
149    5.1
Name: petal_length, Length: 150, dtype: float64, 'versi